File: joiner-test-support/src/test/java/cz/encircled/joiner/test/core/resolver/PredicateConverterTest.java
package cz.encircled.joiner.test.core.resolver;

import cz.encircled.joiner.query.Q;
import cz.encircled.joiner.query.join.J;
import cz.encircled.joiner.test.config.TestConfig;
import cz.encircled.joiner.test.core.AbstractTest;
import cz.encircled.joiner.test.model.QGroup;
import cz.encircled.joiner.test.model.QStatus;
import cz.encircled.joiner.test.model.QUser;
import org.junit.Test;
import org.springframework.test.context.ContextConfiguration;

/**
 * @author Vlad on 04-Aug-16.
 */
@ContextConfiguration(classes = {TestConfig.class})
public class PredicateConverterTest extends AbstractTest {

    @Test
    public void basicTest() {
        joiner.find(Q.from(QGroup.group)
                .joins(J.left(QStatus.status), J.left(QUser.user1).nested(J.left(QStatus.status)))
                .where(QGroup.group.statuses.any().id.eq(2L)));
    }

}

File: joiner-core/src/test/java/cz/encircled/joiner/test/core/resolver/PredicateConverterTest.kt
package cz.encircled.joiner.test.core.resolver

import cz.encircled.joiner.query.Q
import cz.encircled.joiner.query.join.J
import cz.encircled.joiner.test.config.TestConfig
import cz.encircled.joiner.test.core.AbstractTest
import org.junit.Test
import org.springframework.test.context.ContextConfiguration

/**
 * @author Vlad on 04-Aug-16.
 */
@ContextConfiguration(classes = arrayOf(TestConfig::class))
class PredicateConverterTest : AbstractTest() {

    @Test
    fun basicTest() {
        joiner!!.find(Q.from<Any>(QGroup.group).joins(J.left(QStatus.status), J.left(QUser.user1).nested(J.left(QStatus.status))).where(QGroup.group.statuses.any().id.eq(2L)))
    }

}
